### DISCUSSION TITLE
Troubleshoot connecting slave inverter

### DISCUSSION THREAD

**Discussion by JohnBenson00** [wlcrs/huawei_solar#601] [Connecting to the inverter]
After using my 2 inverters for one month now, I've decided that i wanted to integrate home assistant in my ecosystem.
The 2 inverters I own are: 

master: sun-2000 5 ktl l1
and slave: sun-2000 6 ktl l1

slave is connected to master with a 2 wire cable.
I successfully connected the huawei solar integration to the master with TCP, (using the WIFI provided by the master inverter SUN-XX...) but failed reaching slave inverter. After reading the documentation ( https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-daisy-chained-inverters ) I found out that this will not work: "Other options (like the Modbus TCP connection available when connecting to the SUN2000-xxxxxxxxx AP of the master inverter) do not allow to query slave inverters". I really want to get the info from these 2 chained inverters so I found out that my other 2 options are: TCP with dongle ( I don't own a dongle ) or RTU through the RS485A1 and RS485B1 pins on the COM-port  (_this is not verified in practice. and that's what really bothers me_)

Please correct me if I said something wrong. 
What should I do?

**Comment by Roving-Ronin** [comment]
@JohnBenson00 
Simplest to either just go and acquire an sDongle (FE/WiFi), my understanding is that Huawei sells the same product globally and all the residential L1 / M1 inverters are shipped WITH a bundled sDongle for each inverter, so you should have 2 !

For example:

https://osw.energy/au/products/Huawei-Hybrid-L1-5-0kW-1-Phase-2-MPPT-w-wifi-2873  and  https://osw.energy/au/products/Huawei-Hybrid-L1-6-0kW-1-Phase-2-MPPT-w-wifi-2874   (note the line that says:  "WiFi  |  w/ wifi", that means it comes with a WiFi/FE dongle in the box). Might be worth confirming for your country and start asking some polite questions of the reseller / installer, the other 'trick' usually seen is installers selling an inverter without the consumption/smart meter that is required to all the user to see overall kWh consumed (solar+grid), kWh exported to grid etc.  Lowers there quote maybe AUD$200 to 'win' the job, later costs AUD$500 odd to get retrofitted by an electrician.

They ship like this as the intent is that all inverters should be connected to FusionSolar portal that allows Huawei to then globally monitor them for warranty/remote support (as required) and to also allow end users to have their power production / usage / export data available.

If you don't have the meter, here is a link to what the 1p version is, also called a DDSU-666H (DTSU-666H is the 3p version).
https://osw.energy/au/products/Huawei-Smart-Power-Sensor---Single-Phase---RS485-residential-2365

WiFi/FE dongle is this, in AU is costs about AUD$99 (if your free ones break):
https://osw.energy/au/products/HUW-Smart-Dongle-wifi,-Support-wifi-and-Lan-2799

re your last option of connecting to the RS485 ports, that will stop being able to connect to FusionSolar (making it harder to then easily pull down updates and get remote support from Huawei) and still requires a modbus to tcp appliance (that becomes the modbus master, thereby stopping the dongle from working btw) such as the Elfin device mention in 'how to' guides.
